From d03aa37ffe3cae297813d94b3517baea82d7a19d Mon Sep 17 00:00:00 2001
From: Oliver Eftevaag <oliver.eftevaag@qt.io>
Date: Tue, 23 Sep 2025 19:11:35 +0200
Subject: [PATCH 089/239] FontDialogContent: Comply with accessibility insights
 for Windows

Accessibility Insights for Windows is a Windows application that will
check the UI accessibility tree for any potential issues or
anti-patterns.

The tool would complain about the FontDialog not giving the text fields
for font family, style and size an accessibility name.

In addition, the sample text edit was exposed as accessible, even
though it's an non-interactive part of the UI. We can simply hide it
from screen readers.

Pick-to: 6.8
Change-Id: Ice96528c3bcf018fabd95d5aaddd644b41bdfca1
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit 09e2e1951e35e14c8b1ed004b98334fd6063dec1)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../quickdialogsquickimpl/qml/FontDialogContent.qml          | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/quickdialogs/quickdialogsquickimpl/qml/FontDialogContent.qml b/src/quickdialogs/quickdialogsquickimpl/qml/FontDialogContent.qml
index 8b71ad68e9..b8fe859c4e 100644
--- a/src/quickdialogs/quickdialogsquickimpl/qml/FontDialogContent.qml
+++ b/src/quickdialogs/quickdialogsquickimpl/qml/FontDialogContent.qml
@@ -36,6 +36,7 @@ GridLayout {
             readOnly: true
             Layout.fillWidth: true
             focus: true
+            Accessible.name: qsTr("Font family")
         }
         Frame {
             Layout.fillWidth: true
@@ -85,6 +86,7 @@ GridLayout {
             objectName: "styleEdit"
             readOnly: true
             Layout.fillWidth: true
+            Accessible.name: qsTr("Font style")
         }
         Frame {
             Layout.fillWidth: true
@@ -134,6 +136,7 @@ GridLayout {
                 bottom: 1
                 top: 512
             }
+            Accessible.name: qsTr("Font point size")
         }
         Frame {
             Layout.fillWidth: true
@@ -231,6 +234,8 @@ GridLayout {
             anchors.centerIn: parent
             readOnly: true
             color: palette.text
+            focusPolicy: Qt.NoFocus
+            Accessible.ignored: true
         }
     }
 }
-- 
2.51.2

